Lib/test/test_math.py

```diff
@@ -623,6 +624,98 @@ def testFmod(self):
         self.assertEqual(math.fmod(0.0, NINF), 0.0)
         self.assertRaises(ValueError, math.fmod, INF, INF)
 
+    def test_fmax(self):
+        self.assertRaises(TypeError, math.fmax)
+        self.assertRaises(TypeError, math.fmax, 'x', 'y')
+
+        self.assertEqual(math.fmax(0., 0.), 0.)
+        self.assertEqual(math.fmax(0., -0.), 0.)
+        self.assertEqual(math.fmax(-0., 0.), 0.)
+
+        self.assertEqual(math.fmax(1., 0.), 1.)
+        self.assertEqual(math.fmax(0., 1.), 1.)
+        self.assertEqual(math.fmax(1., -0.), 1.)
+        self.assertEqual(math.fmax(-0., 1.), 1.)
+
+        self.assertEqual(math.fmax(-1., 0.), 0.)
+        self.assertEqual(math.fmax(0., -1.), 0.)
+        self.assertEqual(math.fmax(-1., -0.), -0.)
+        self.assertEqual(math.fmax(-0., -1.), -0.)
+
+        for x in [NINF, -1., -0., 0., 1., INF]:
+            self.assertFalse(math.isnan(x))
+
+            with self.subTest("math.fmax(INF, x)", x=x):
+                self.assertEqual(math.fmax(INF, x), INF)
+            with self.subTest("math.fmax(x, INF)", x=x):
+                self.assertEqual(math.fmax(x, INF), INF)
+
+            with self.subTest("math.fmax(NINF, x)", x=x):
+                self.assertEqual(math.fmax(NINF, x), x)
+            with self.subTest("math.fmax(x, NINF)", x=x):
+                self.assertEqual(math.fmax(x, NINF), x)
```

Comment:
The traceback contains the line causing the assertion failure, I don't think that it's worth it to repeat it it subTest():

```suggestion
            with self.subTest(x=x):
                self.assertEqual(math.fmax(INF, x), INF)
                self.assertEqual(math.fmax(x, INF), INF)
                self.assertEqual(math.fmax(NINF, x), x)
                self.assertEqual(math.fmax(x, NINF), x)
```